dspace-uclouvain/src/main/java/org/dspace/uclouvain/external/dilbert/client/DilbertClient.java

```diff
@@ -26,31 +28,70 @@ public DilbertClient(UCLouvainAuthorAuthorityAPIConfiguration uclouvainAuthority
-     * It returns a list of student information
+     * Search by student first name or/and second name and filtering by {filter} (ex: filter="student").
+     * It returns a list of student information.
      * 
-     * @param term
-     * @param filter
-     * @return: A list of dial peron information 
+     * @param term: The name and/or second name of the student.
+     * @return: A list of dial person information.
      */
     public DialPerson[] getStudentByTermWithFilter(String term) {
         // GET {apiUrl}/searchAuthorDrupal.php?term={term}&filter={filter}
-        String filter = "&filter=" + uclouvainAuthorityConfiguration.getFilter();
-        String url = "/searchAuthorDrupal.php?term=" + term + filter;
+        String filter = "&filter=" + uclouvainAuthorityConfiguration.getAuthorFilter();
+        return this.getDialPersonByTermWithFilter(term, filter);
+    }
+
+    /** 
+     * Search by promoter first name or/and second name and filtering by {filter}..
+     * It returns a list of promoter information.
+     * 
+     * @param term: The name and/or second name of the promoter.
+     * @return: A list of dial person information.
+     */
+    public DialPerson[] getPromoterByTermWithFilter(String term) {
+        String filter = "&filter=" + uclouvainAuthorityConfiguration.getPromoterFilter();
+        return this.getDialPersonByTermWithFilter(term, filter);
+    }
+
+    /** 
+     * Generic method to call the Dilbert API with a given term && filter.
+     * 
+     * @param term: The name and/or second name of the person.
+     * @param filter: The filter which indicate the type of person to extract (ex: student, promoter...)
+     * @return: A list of dial person information.
+     */
+    private DialPerson[] getDialPersonByTermWithFilter(String term, String filter) {
+        String cleanedTerm = this.encodeURL(term);
+        String url = "/searchAuthorDrupal.php?term=" + cleanedTerm + filter;
 
         DialPerson[] dialPerson = {};
 
         try {
             HttpResponse<String> response = httpClient.get(url);
             dialPerson = new GenericResponse(response).extractJsonResponseDataToClass(null, DialPerson[].class);
         }
-        catch(IOException | InterruptedException e) {
+        catch(Exception e) {
             System.err.println(e.getClass().getSimpleName() + "while fetching data :: " + e.getMessage());
             e.printStackTrace(System.err);
         }
         return dialPerson;
     }
 
+    // Utils
+    /**
+     * Encode URL to handle escaping characters like spaces.
+     * 
+     * @param url: The string url to encode.
+     */
+    private String encodeURL(String url) {
+        try {
+            return URLEncoder.encode(url, StandardCharsets.UTF_8.toString());
+        }
+        catch(UnsupportedEncodingException e) {
+            System.out.println("Error while encodingURL ::" + e.getMessage());
```

Comment:
Use "System.err.println" at minimum. It should be better to use `Logger` (Log4J ?) library to log these message if needed.